README.md

```diff
@@ -36,80 +36,172 @@ spec:
+Each configMap additionally needs to have `cidr-filtering-cni` label to be enabled for the cidr-filtering-cni plugin.
+
+The following manifest is an example of a minimal configMap, that doesn't yet allow any traffic. It's given as an example, since denying all traffic is done by default, it's
+not needed to be explicitly created.
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: deny-all
+  labels:
+    cidr-filtering-cni: ""
+    br1-with-cidr-filtering: ""
+data:
+  config.json: |
+    {
+      "ingress": {},
+      "egress": {}
+    }
+```
+
+Notice the labels of the configMap.
+The first label, `cidr-filtering-cni` ensures that the configMap configuration is collected by the plugin, as
+the plugin only lists configMaps with this label.
+The second label, `br1-with-cidr-filtering` refers to the NetworkAttachmentDefinition spec name, in the cni configuration JSON.
+
+
+> **_NOTE:_**  A configMap may refer to many NetworkAttachmentDefinition specs. The order in which configMaps are process is not defined.
+
+
+To allow a pod to communicate to external endpoints, let's create the following configMap, that allows pod to reach all ports
+on a local network:
+
+```yaml
 ---
 kind: ConfigMap
 apiVersion: v1
 metadata:
-  name: local-network
+  name: egress-local-network
   namespace: default
   labels:
     cidr-filtering-cni: ""
     br1-with-cidr-filtering: ""
 data:
   config.json: |
     {
-      "ingress": {
+      "egress": {
         "subnets": [
           {
             "subnet": {
-              "cidr": "192.168.66.0/24",
+              "cidr": "192.168.0.0/16",
               "except": [
-                "192.168.66.105"
+                "192.168.150.0/24",
+                "192.168.151.151"
               ]
             }
           }
+        ],
+        "ports": [
+          {
+            "protocol": "TCP",
+            "port": "80"
+          },
+          {
+            "protocol": "tcp",
+            "port": "8080"
+          }
         ]
-      },
-      "egress": {
+      }
+    }
+```
+
+The `egress.subnets` attribute allows pod to reach to any IP address in subnet `192.168.0.0/16`, except for subnet `192.168.150.0/24` and `192.168.151.151`.
+The `egress.ports` attribute allows pod to reach ports 80, and 8080 over TCP protocol.
+
+To allow any webserver responses to be allowed to reach the pod, create the following configmap:
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: ingress-local-network
+  labels:
+    cidr-filtering-cni: ""
+    br1-with-cidr-filtering: ""
+data:
+  config.json: |
+    {
+      "ingress": {
         "subnets": [
           {
             "subnet": {
-              "cidr": "192.168.66.0/16",
-              "except": [
-                "192.168.1.0/24",
-                "192.168.2.0/24"
-              ]
+              "cidr": "192.168.0.0/16"
             }
           }
-        ]
+        ],
+        "ports": []
```

Comment:
This is the default, no?

Reply:
This allows all ports, which is not default.
Default is denying all, which is the same as not specifying the ports key at all.
This is the same behaviour as NetworkPolicy has.

Reply:
And with subnets the default is the opposite? I see that the DHCP example only mentions ports, so I would expect it does not allow any subnets, as no ports attribute would allow no ports?

I'm ok with this if it copies the NetworkPolicy behavior, it is just confusing to me.

Reply:
The default is the same, with nothing being allowed by default.

We need a way to express in a policy to
1) not change a subnet/port allow list
2) allow all ports/subnets

1) is done by not specifying `subnets` or `ports` array
2) is done by specifying `ports: []` or `subnets: []`

```
... Items in this list are combined using a logical OR operation. If this field is empty or missing, this rule matches all destinations (traffic not restricted by destination). If this field is present and contains at least one item, this rule allows traffic only if the traffic matches at least one item in the to list.
```

The case for empty list is the same as is done in network policies, the difference is that ommiting the key means no-op in this CNI, but means `allow-all` in NetworkPolicy

Reply:
> the difference is that ommiting the key means no-op in this CNI, but means allow-all in NetworkPolicy

Why are we different in that?

Reply:
Nevermind, we don't have to be the same.

Reply:
Addressed this in https://github.com/rhrazdil/cidr-filtering-cni/pull/13/commits/c5ac9f84f954769faa83bb0c2fcfbb98ebb0c0b2